velox/functions/prestosql/tests/JsonFunctionsTest.cpp

```diff
@@ -100,6 +101,59 @@ TEST_F(JsonFunctionsTest, jsonFormat) {
   facebook::velox::test::assertEqualVectors(expected, result);
 }
 
+TEST_F(JsonFunctionsTest, jsonParse) {
+  const auto jsonParse = [&](std::optional<std::string> value) {
+    return evaluateOnce<StringView>("json_parse(c0)", value);
+  };
+
+  EXPECT_EQ(jsonParse(std::nullopt), std::nullopt);
+  EXPECT_EQ(jsonParse(R"(true)"), "true");
+  EXPECT_EQ(jsonParse(R"(null)"), "null");
+  EXPECT_EQ(jsonParse(R"(42)"), "42");
+  EXPECT_EQ(jsonParse(R"("abc")"), "\"abc\"");
+  EXPECT_EQ(jsonParse(R"([1, 2, 3])"), "[1, 2, 3]");
+  EXPECT_EQ(jsonParse(R"({"k1":"v1"})"), "{\"k1\":\"v1\"}");
+  EXPECT_EQ(jsonParse(R"(["k1", "v1"])"), "[\"k1\", \"v1\"]");
+  EXPECT_THROW(jsonParse(R"({"k1":})"), VeloxUserError);
+  EXPECT_THROW(jsonParse(R"({:"k1"})"), VeloxUserError);
+  EXPECT_THROW(jsonParse(R"(not_json)"), VeloxUserError);
+
+  auto data = makeRowVector({makeConstant(R"("k1":)", 2)});
+  EXPECT_THROW(evaluate("json_parse(c0)", data), VeloxUserError);
+
+  VELOX_ASSERT_THROW(jsonParse(R"(not_json)"), "expected json value");
+
+  data = makeRowVector({makeConstant(R"("k1":)", 2)});
+  VELOX_ASSERT_THROW(
```

Comment:
For each failing test case, add another one with TRY and verify there is not error, but result is NULL.

Please, add a test case where some rows fail parsing, but others succeed. Verify results with and without TRY.